src/Cli/dotnet/UILanguageOverride.cs

```diff
@@ -61,6 +82,11 @@ private static CultureInfo GetOverriddenUILanguage()
                 catch (CultureNotFoundException) { }
             }
 
+            if (!CultureInfo.InstalledUICulture.TwoLetterISOLanguageName.Equals("en", StringComparison.InvariantCultureIgnoreCase))
```

Comment:
what you are trying to do here? CultureInfo.InstalledUICulture not necessary will tell the user UI language. I don't think this logic correct and I am not sure what you are trying to achieve here.


[https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Globalization/CultureInfo.cs,422](https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Globalization/CultureInfo.cs,422)

Reply:
We are getting the OS System Language here to check if it is english

Reply:
What you mean by the system language? User locale? UI locale? system locale? and if it is English, why you force the UI langue to English at that time? 

InstalledUILanguage returns the user locale which is not necessarily the default UI language on the system. I am wondering why you don't just run with whatever UI language set by the user. Why do you want to change this setting even when not using any config switch to override the user settings?

Reply:
Thanks for taking the time to review!

Reply:
I think the intent here is to set the output encoding to UTF8 if we're using a non-English language, either because it's the default or because it's overridden using `DOTNET_CLI_UI_LANGUAGE` or `VSLANG` environment variables.

Reply:
By system language I mean this:
![image](https://user-images.githubusercontent.com/23152278/211685732-cec99391-7f3e-40e1-922f-e6f9792b6467.png)

We are setting the encoding if it is not english to support non english chars. If the system language is JP then dotnet should also be in JP and shouldn't show ??? because it would be in US english chcp page without this change. If my system is set to JP and I don't have english installed on my system dotnet shouldn't be in english by default.

Reply:
I hope we focus more here on the user scenarios to know what exactly we need to achieve. If it is around the console encoding, I am not seeing any reason we need to change anything handling User/UI language.

Reply:
> I am not seeing any reason we need to change anything handling User/UI language

Maybe this is referring to the code above... but the InstalledUICulture is referring to the OS Display Language. https://learn.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo.installeduiculture?view=netframework-4.8 This is focused on a specific user scenario, which is what's stated above: currently if my OS is in a not english language `dotnet` will still be in english, this makes it so it uses the language of the OS.

Reply:
`InstalledUICulture` is not doing what you are looking for. Imagine you install Windows machine which has UI language like Japanese and setting the user locale to English. `InstalledUICulture` will have English at that time and not Japanese. If you want get the UI language, just use `CultureInfo.CurrentUICulture`.

Reply:
Thanks again for taking the time to review and explain 😄